cmd/controller-manager/app/controller-manager.go

```diff
@@ -117,6 +130,14 @@ func Run(opts *options.Options) error {
 		klog.Fatalf("Invalid Feature Gate: %v", err)
 	}
 
+	if len(masterURL) == 0 && len(kubeconfig) == 0 && len(opts.Config.FederationNamespace) == 0 {
+		fedName, err := defaultNamespace()
+		if err != nil {
+			klog.Fatalf("The controller-manager is not running inside the cluster. Use --federation-namespace flag to manually set the fedeation-namespace")
```

Comment:
The error should be logged and the namespace flag should be checked outside of this conditional, i.e.

```golang
if len(masterURL)... {
  fedNamespace, err := ...
  if err != nil {
    // Include the err in the log message to aid in troubleshooting
    klog.Fatalf("An error occurred while attempting to discover the federation namespace from the service account: %v", err)
  } 
}

// Regardless of whether in-cluster discovery was attempted, validate that a federation namespace is configured 
if len(opts.Config.FederationNamespace) == 0 {
  klog.Fatalf("The federation namespace must be specified via --federation-namespace")
} 
```

Errors should almost always be logged unless you have a good reason not to.  It's entirely possible the error will convey information vital to a troubleshooting exercise.

Reply:
ack